geniusyield-market-maker/src/GeniusYield/MarketMaker/Utils.hs

```diff
@@ -53,3 +66,132 @@ instance HasPartialOrderNftScript DEXInfo where
 
 instance HasPartialOrderConfigAddr DEXInfo where
   getPartialOrderConfigAddr = dexPartialOrderConfigAddr
+
+
+-------------------------------------------------------------------------------
+-- Common Price Resolution
+-------------------------------------------------------------------------------
+
+-- | Generic time resolution for OHLC Candles
+data CommonResolution = CRes5m | CRes15m | CRes30m | CRes1h | CRes4h
+                      | CRes1d | CRes1w  | CRes1mo
+                      deriving stock (Eq, Ord, Show, Generic)
+
+instance FromJSON CommonResolution
```

Comment:
Needs modification to match convention (snake case).

Reply:
And to remove `CRes` in json value, look at `StripPrefix`.

Reply:
I am also fine with getting rid of common resolution altogether as having it implies an assumption that all price providers must have resolution.

Reply:
Originally I thought that having a common resolution was crucial.  But testing shows that last closing price appears to be independent of resolution.

Reply:
Update:  as discussed, further experimentation seems to show that closing price does depend on the resolution, at least for Taptools.  For example, there is mismatch between 1 hour and 1 day resolution for "iBTC".  Hence I am keeping "common resolution" for now.

Reply:
> And to remove `CRes` in json value, look at `StripPrefix`.

My current solution is:
```haskell
instance FromJSON CommonResolution where
    parseJSON (String v) =
      let kvm = Map.fromList [ ("5m", CRes5m), ("15m", CRes15m), ("30m", CRes30m), ("1h", CRes1h),
                               ("4h", CRes4h), ("1d", CRes1d), ("1w", CRes1w), ("1mo", CRes1mo) ]
      in case Map.lookup v kvm of
        Nothing   -> typeMismatch "CommonResolution" ""
        Just cres -> pure cres
    parseJSON invalid    = typeMismatch "CommonResolution" invalid
```

I tried 
```haskell
  deriving (FromJSON) via CustomJSON '[FieldLabelModifier '[StripPrefix Abc]] CommonResolution
```
but `StripPrefix` seems to only work in the context of a record datatype.

Reply:
> Needs modification to match convention (snake case).

Since we are not deriving `ToJSON`, I don't think we need to worry about snake case here.

Reply:
Got rid of "common resolution"!